main.go

```diff
@@ -75,7 +93,7 @@ func (s *service) start() error {
 	listener, err := net.Listen("tcp", s.conf.Listen)
 
 	if err != nil {
-		return err
+		log.Fatalln(err)
```

Comment:
Read the `main` function code - errors returned from this function are already logged - so no important error messages will be hidden if you just return the error.

```go
startErr := srv.start()
if startErr != nil {
	mainlog.Printf("Could not start service, reason: %q", startErr.Error())
}
```